Update `hpc-repo-tools`

Onitoxan opened the pull request:
## Nature of PR

- Bug-fix
- Hot-fix
- Feature
- Testing
- 🟢 Rewrite
- 🟢 Dependencies
- CI


## Description

In this PR we update to `hpc-repo-tools` `v5.7.1` where we fixed `unicorn` ESLint rules not working in our projects. Beside updating, I refactored all code that were not complying to the rules

## How to test changes

I would say is straight forward

Onitoxan committed:
Update `hpc-repo-tools` and peer deps

Onitoxan committed:
Update code to follow ESLint rules

unocha-hpc reviewed (approved):
Checks have passed and this pull request is ready for manual review

Pl217 committed:
Remove unused parameter

Pl217 committed:
Change the rule being suppressed

The rule `@typescript-eslint/no-empty-interface`
has been deprecated in favour of the more comprehensive
`@typescript-eslint/no-empty-object-type` rule.

We ignore it anyway, but since rule was deprecated,
warning "Unused eslint-disable directive" was reported

unocha-hpc reviewed (approved):
Checks have passed and this pull request is ready for manual review

Pl217 reviewed (approved):
I have pushed a couple of commits to also get rid of a few warnings. Will rebase on top of latest `develop` and merge.

unocha-hpc reviewed (approved):
Checks have passed and this pull request is ready for manual review
